### PR TITLE
Rename IPFamilyPolicyType => IPFamilyPolicy

### PR DESCRIPTION
**Pull request by thockin**
Just a bit of redundancy removal.  Old name left for compat.

/kind cleanup

```release-note
NONE
```
 /sig network

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -3736,27 +3736,27 @@ const (
 	IPv6Protocol IPFamily = "IPv6"
 )
 
-// IPFamilyPolicyType represents the dual-stack-ness requested or required by a Service
-type IPFamilyPolicyType string
```
Comment (robscott) [inline pkg/apis/core/types.go:3740]: @thockin Your PR message says that the old name was left for compat but I can't see it being left behind anywhere. I don't think that's necessary, just making sure this PR isn't missing anything you intended to included.

**Review by robscott — COMMENTED** [review]
This LGTM, just one question.

File: pkg/apis/core/types.go (hunk 1 above)
Comment (thockin) [reply pkg/apis/core/types.go:3740]: Look in the v1 file:

```diff
+// for backwards compat
+type IPFamilyPolicyType = IPFamilyPolicy
```

**Comment by thockin** [comment]
I realized I forgtot to `+enum` the alias.  Re-pushed.

File: pkg/apis/core/types.go (hunk 1 above)
Comment (robscott) [reply pkg/apis/core/types.go:3740]: Thanks!

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**